src/vsock/src/packet.rs

```diff
@@ -0,0 +1,343 @@
+#![deny(missing_docs)]
+
+use super::vhu_vsock::{Error, Result};
+use byteorder::{ByteOrder, LittleEndian};
+use virtio_queue::DescriptorChain;
+use vm_memory::{
+    GuestAddress, GuestAddressSpace, GuestMemory, GuestMemoryAtomic, GuestMemoryLoadGuard,
+    GuestMemoryMmap,
+};
+
+// vsock packet header size when packed
```

Comment:
As a TODO item maybe we can replace these with a `bindgen` generated struct, but for now it should be fine.